seth.sh

```diff
@@ -11,32 +11,55 @@ EOF
 
 set -e
 
+# Ensure we have root permission
 if [ "$(id -u)" != "0" ]; then
     echo "This script must be run as root" 1>&2
     exit 1
 fi
 
-if [ "$#" -ne 4 -a "$#" -ne 5 ]; then
-    cat << EOF
-Usage:
-$0 <INTERFACE> <ATTACKER_IP> <VICTIM_IP> <GATEWAY_IP|HOST_IP> [<COMMAND>]
-EOF
+# Ensure we have all the required arguments
+if [ "$#" -ne 4 -a "$#" -ne 5 ];
+then
+    echo "Usage:"
+    echo "$0 <INTERFACE> <ATTACKER_IP> <VICTIM_IP> <GATEWAY_IP|HOST_IP> [<COMMAND>]"
     exit 1
 fi
 
-for com in tcpdump arpspoof openssl iptables ; do
-    command -v "$com" >/dev/null 2>&1 || {
-        echo >&2 "$com required, but it's not installed.  Aborting."
-        exit 1
-    }
-done
+# Get OS Name
+OS=$(uname -s)
+
+# Setup requirements based on OS
+# Darwin (MacOS) uses pf as the packet filter module
+# while Linux uses iptables
+#
+# Note: dsniff seems to be abandoned
+# TODO: implement own arpspoofer, or relay on something currently maitained eg: bettercap
+# TODO: Allow use of a custom openssl version by path
+if [ "$OS" == "Darwin" ];
+then
+    echo "[*] Darwin OS detected, switching from iptables to pf"
+
+    for com in tcpdump arpspoof openssl pfctl ; do
+        command -v "$com" >/dev/null 2>&1 || {
+            echo >&2 "$com required, but it's not installed.  Aborting."
+            exit 1
+        }
+    done
+else
+    for com in tcpdump arpspoof openssl iptables ; do
+        command -v "$com" >/dev/null 2>&1 || {
+            echo >&2 "$com required, but it's not installed.  Aborting."
+            exit 1
+        }
+    done
+fi
```

Comment:
I think I'd like this change better, if we only changed the list dependencies in the conditional. I like to avoid duplicate code.

Reply:
Sure, do you prefer to parametrize the whole list, or only the NETFILTER_INTERPRETER?
like:
```suggestion
if [ "$OS" == "Darwin" ];
then
    NETFILTER_INTERPRETER=pfctl
else
    NETFILTER_INTERPRETER=iptables
fi
```
or
```suggestion
if [ "$OS" == "Darwin" ];
then
    DEPS_LIST="pfctl openssl arpspoof tcpdump"
else
    DEPS_LIST="iptables openssl arpspoof tcpdump"
fi
```